Assets/MixedRealityToolkit/Providers/UnityInput/MouseController.cs

```diff
@@ -65,32 +64,39 @@ public void Update()
                 return;
             }
 
-            if (InputSource.Pointers[0].BaseCursor != null)
-            {
-                controllerPose.Position = InputSource.Pointers[0].BaseCursor.Position;
-                controllerPose.Rotation = InputSource.Pointers[0].BaseCursor.Rotation;
-            }
-
-            mouseDelta.x = -UInput.GetAxis("Mouse Y");
-            mouseDelta.y = UInput.GetAxis("Mouse X");
-            MixedRealityToolkit.InputSystem?.RaiseSourcePositionChanged(InputSource, this, mouseDelta);
-            MixedRealityToolkit.InputSystem?.RaiseSourcePoseChanged(InputSource, this, controllerPose);
-            MixedRealityToolkit.InputSystem?.RaiseSourcePositionChanged(InputSource, this, UInput.mouseScrollDelta);
-
             for (int i = 0; i < Interactions.Length; i++)
             {
+                
+
                 if (Interactions[i].InputType == DeviceInputType.SpatialPointer)
                 {
+                    // add mouse delta as rotation
+                    var mouseDeltaRotation = Vector3.zero;
+                    mouseDeltaRotation.x += -UInput.GetAxis("Mouse Y");
+                    mouseDeltaRotation.y += UInput.GetAxis("Mouse X");
+                    IMixedRealityMousePointer mousePointer = (IMixedRealityMousePointer)InputSource.Pointers[0];
+                    if (mousePointer != null)
+                    {
+                        mouseDeltaRotation *= mousePointer.Speed;
+                    }
+
+                    MixedRealityPose controllerPose = MixedRealityPose.ZeroIdentity;
+                    controllerPose.Rotation = Quaternion.Euler(mouseDeltaRotation);
                     Interactions[i].PoseData = controllerPose;
```

Comment:
>Interactions[i].PoseData = controllerPose [](start = 20, length = 41)

We seem to be storing just a rotation delta here. It should be a proper world space pose instead. The idea is to use that directly in the mouse pointer as ray cast origin.

Reply:
The problem I have here is that I don't have access to the previous mouse position - I shouldn't query the pointer so I can't provide the position here if we want to keep using the delta